app/code/Magento/Newsletter/Model/ResourceModel/Subscriber.php

```diff
@@ -110,6 +110,30 @@ public function loadByEmail($subscriberEmail)
         return $result;
     }
 
+    /**
+     * Load subscriber from DB by email and storeId
+     *
+     * @param string $subscriberEmail
+     * @param int $storeId
+     * @return array
+     */
+    public function loadByEmailAndStore($subscriberEmail, $storeId)
```

Comment:
Unfortunately this class marked as @api, so according to https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/ we can't new public methods into it